python/console/console_sci.py

```diff
@@ -631,6 +631,7 @@ def _deleteItem(self):
             item = itemsSelected[0].row()
             # Remove item from the command history (just for the current session)
             self.parent.history.pop(item)
-            self.parent.historyIndex -= 1
+            self.parent.softHistoryIndex -= 1
+            self.parent.syncSoftHistory()
```

Comment:
`syncSoftHistory()` completely rewrites the soft history, so modifying the `softHistoryIndex` beforehand makes no change.
Furthermore, by completely rewriting the soft history, any modified soft history entry is reset, including the last entry ie the one typed but not executed.
Example:
- type `command1`, don't hit enter
- press up, now `old_command2` shows
- press ctrl+shift+space, locate `old_command17` and press del
- `old_command17` no longer exists on the history but `command1` has been replaced in the soft history by `old_command2`

Reply:
@uclaros correct, ```self.parent.softHistoryIndex -= 1``` is not necessary!

And yes, if you do not hit the Enter key it is not stored in the history, so it is skipped. By deleting entry in history dialog it is necessary to update softHistory in order to have the same history by hitting key up/down.

@nirvn do you have an alternative approach to solve the issue?

Reply:
Something like:
```
self.parent.softHistory.pop(item)
if item < self.parent.softHistoryIndex:
    self.parent.softHistoryIndex -= 1
self.parent.setText(self.parent.softHistory[self.parent.softHistoryIndex])
self.parent.move_cursor_to_end()
```

Reply:
@uclaros if the proposed changes work fine for you please issue a PR with the changes. I done a quick test and work fine for me. Thanks. I will close this PR.

Also, I think we have to introduce the possibility to delete permanently the entry being deleted from history dialog.

Reply:
What's the status here @slarosa @uclaros ? Should this PR be closed or left to be updated?